crates/core/src/kvs/mem/mod.rs

```diff
@@ -142,16 +158,23 @@ impl super::api::Transaction for Transaction {
 	where
 		K: KeyEncode + Sprintable + Debug,
 	{
-		// Memory does not support versioned queries.
-		if version.is_some() {
-			return Err(Error::UnsupportedVersionedQueries);
-		}
 		// Check to see if transaction is closed
 		if self.done {
 			return Err(Error::TxFinished);
 		}
-		// Check the key
-		let res = self.inner.get(&key.encode()?)?.is_some();
+		// Get the arguments
+		let key = key.encode_owned()?;
+		// Execute on the blocking threadpool
+		let res = affinitypool::spawn_local(|| -> Result<_, Error> {
```

Comment:
I don't think any of the in-mem ops needs affinity pool at all, I understand it would be need for skv, but what benefit would it bring here given that all ops are in mem?

Reply:
Yeah that's a good point!